packages/jaeger-ui/src/components/TracePage/TraceGraph/TraceGraph.js

```diff
@@ -0,0 +1,328 @@
+        const cdr = new DRange(m.span.startTime, m.span.startTime + m.span.duration).intersect(
+          this.getChildOfDrange(m.span.spanID)
+        );
+        return p + cdr.length;
+      }, 0);
+      const stime = ntime - childDurationsDRange;
+      const nd = {
+        count: n.members.length,
+        errors: numErrors,
+        time: ntime,
+        percent: 100 / this.props.trace.duration * ntime,
+        selfTime: stime,
+        percentSelfTime: 100 / ntime * stime,
+      };
+      // eslint-disable-next-line no-param-reassign
+      n.data = nd;
+    });
+    return traceDag;
+  }
+
+  getChildOfDrange(parentID: string): number {
+    const childrenDrange = new DRange();
+    this.getChildOfSpans(parentID).forEach(s => {
+      // -1 otherwise it will take for each child a micro (incluse,exclusive)
+      childrenDrange.add(s.startTime, s.startTime + (s.duration <= 0 ? 0 : s.duration - 1));
+    });
+    return childrenDrange;
+  }
+
+  getChildOfSpans(parentID: string): Span[] {
+    if (!this.parentChildOfMap) {
+      this.parentChildOfMap = {};
+      this.props.trace.spans.forEach(s => {
+        if (s.references) {
+          // Filter for CHILD_OF we don't want to calculate FOLLOWS_FROM (prod-cons)
+          const parentIDs = s.references.filter(r => r.refType === 'CHILD_OF').map(r => r.spanID);
+          parentIDs.forEach((pID: string) => {
+            this.parentChildOfMap[pID] = this.parentChildOfMap[pID] || [];
+            this.parentChildOfMap[pID].push(s);
+          });
+        }
+      });
+    }
+    return this.parentChildOfMap[parentID] || [];
+  }
+
+  toggleNodeMode(newMode: string) {
+    this.setState({ mode: newMode });
+  }
+
+  showHelp = () => {
+    this.setState({ showHelp: true });
+  };
+
+  closeSidebar = () => {
+    this.setState({ showHelp: false });
+  };
+
+  render() {
+    const { headerHeight, trace } = this.props;
+    const { showHelp, mode } = this.state;
+    if (!trace) {
+      return <h1 className="u-mt-vast u-tx-muted ub-tx-center">No trace found</h1>;
+    }
+
+    // Caching edges/vertices so that DirectedGraph is not redrawn
+    let ev = this.cache;
+    if (!ev) {
+      const traceDag = this.calculateTraceDag();
+      const nodes = [...traceDag.nodesMap.values()];
+      ev = convPlexus(traceDag.nodesMap);
+      ev.edges = extendFollowsFrom(ev.edges, nodes);
+      this.cache = ev;
+    }
+
+    return (
+      <div className="TraceGraph--graphWrapper" style={{ paddingTop: headerHeight + 49 }}>
+        <DirectedGraph
+          minimap
+          zoom
+          arrowScaleDampener={0}
+          className="TraceGraph--dag"
+          minimapClassName="TraceGraph--miniMap"
+          layoutManager={this.layoutManager}
+          getNodeLabel={getNodeDrawer(mode)}
+          setOnRoot={classNameIsSmall}
+          setOnEdgePath={setOnEdgePath}
+          edges={ev.edges}
+          vertices={ev.vertices}
+        />
+        <div className="TraceGraph--experimental">
```

Comment:
Looks great. I don't think you need the wrapping `<div>`; the styles on the div can be applied to the `<a>`.